Source/itemdat.cpp

```diff
@@ -5,545 +5,626 @@
+	if (value == "RING") return UITYPE_RING;
+	if (value == "AMULET") return UITYPE_AMULET;
+	if (value == "SKCROWN") return UITYPE_SKCROWN;
+	if (value == "INFRARING") return UITYPE_INFRARING;
+	if (value == "OPTAMULET") return UITYPE_OPTAMULET;
+	if (value == "TRING") return UITYPE_TRING;
+	if (value == "HARCREST") return UITYPE_HARCREST;
+	if (value == "MAPOFDOOM") return UITYPE_MAPOFDOOM;
+	if (value == "ELIXIR") return UITYPE_ELIXIR;
+	if (value == "ARMOFVAL") return UITYPE_ARMOFVAL;
+	if (value == "STEELVEIL") return UITYPE_STEELVEIL;
+	if (value == "GRISWOLD") return UITYPE_GRISWOLD;
+	if (value == "LGTFORGE") return UITYPE_LGTFORGE;
+	if (value == "LAZSTAFF") return UITYPE_LAZSTAFF;
+	if (value == "BOVINE") return UITYPE_BOVINE;
+	if (value == "INVALID") return UITYPE_INVALID;
+	return tl::make_unexpected("Unknown enum value");
+}
+
+tl::expected<ItemSpecialEffect, std::string> ParseItemSpecialEffect(std::string_view value)
+{
+	if (value == "RandomStealLife") return ItemSpecialEffect::RandomStealLife;
+	if (value == "RandomArrowVelocity") return ItemSpecialEffect::RandomArrowVelocity;
+	if (value == "FireArrows") return ItemSpecialEffect::FireArrows;
+	if (value == "FireDamage") return ItemSpecialEffect::FireDamage;
+	if (value == "LightningDamage") return ItemSpecialEffect::LightningDamage;
+	if (value == "DrainLife") return ItemSpecialEffect::DrainLife;
+	if (value == "MultipleArrows") return ItemSpecialEffect::MultipleArrows;
+	if (value == "Knockback") return ItemSpecialEffect::Knockback;
+	if (value == "StealMana3") return ItemSpecialEffect::StealMana3;
+	if (value == "StealMana5") return ItemSpecialEffect::StealMana5;
+	if (value == "StealLife3") return ItemSpecialEffect::StealLife3;
+	if (value == "StealLife5") return ItemSpecialEffect::StealLife5;
+	if (value == "QuickAttack") return ItemSpecialEffect::QuickAttack;
+	if (value == "FastAttack") return ItemSpecialEffect::FastAttack;
+	if (value == "FasterAttack") return ItemSpecialEffect::FasterAttack;
+	if (value == "FastestAttack") return ItemSpecialEffect::FastestAttack;
+	if (value == "FastHitRecovery") return ItemSpecialEffect::FastHitRecovery;
+	if (value == "FasterHitRecovery") return ItemSpecialEffect::FasterHitRecovery;
+	if (value == "FastestHitRecovery") return ItemSpecialEffect::FastestHitRecovery;
+	if (value == "FastBlock") return ItemSpecialEffect::FastBlock;
+	if (value == "LightningArrows") return ItemSpecialEffect::LightningArrows;
+	if (value == "Thorns") return ItemSpecialEffect::Thorns;
+	if (value == "NoMana") return ItemSpecialEffect::NoMana;
+	if (value == "HalfTrapDamage") return ItemSpecialEffect::HalfTrapDamage;
+	if (value == "TripleDemonDamage") return ItemSpecialEffect::TripleDemonDamage;
+	if (value == "ZeroResistance") return ItemSpecialEffect::ZeroResistance;
+	return tl::make_unexpected("Unknown enum value");
+}
+
+tl::expected<item_misc_id, std::string> ParseItemMiscId(std::string_view value)
+{
+	if (value == "NONE") return IMISC_NONE;

```

Comment:
Only comment is that I think it would be good if we changed these to follow CamelCase like the enum classes, even if we haven't done so internally yet. That way it's more consistent and hopefully also stable.

But I think this might also have been the case for some of the other files so I don't think that has to be a blocker for this one.